### PR TITLE
Web Interface i18n

### PR DESCRIPTION
**Pull request by rdmark**
Includes a complete Swedish localization.
To test, set your browser to Swedish locale (e.g. in Firefox, you can set the "browser menu and messages" to Swedish, and the locale accept headers gets set accordingly.)

Known issue: the default strings that are generated by the Dropzone JS  javascript library are not internationalized at this point.

### CODE REVIEW

**Comment by bzeiss** [comment]
@rdmark fresh clone of the git repo, checkout on rdmark_webui_babel. ./easy_install.sh, selected 1), enabled token-based access control: no. After starting up, I get this:
```
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]: ERROR:web:Exception on / [GET]
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]: Traceback (most recent call last):
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]:   File "/home/pi/RASCSI/src/web/venv/lib/python3.9/site-packages/flask/app.py", line 2070, in wsgi_app
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]:     response = self.full_dispatch_request()
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]:   File "/home/pi/RASCSI/src/web/venv/lib/python3.9/site-packages/flask/app.py", line 1515, in full_dispatch_request
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]:     rv = self.handle_user_exception(e)
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]:   File "/home/pi/RASCSI/src/web/venv/lib/python3.9/site-packages/flask/app.py", line 1513, in full_dispatch_request
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]:     rv = self.dispatch_request()
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]:   File "/home/pi/RASCSI/src/web/venv/lib/python3.9/site-packages/flask/app.py", line 1499, in dispatch_request
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]:     return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]:   File "/home/pi/RASCSI/src/web/web.py", line 134, in index
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]:     return render_template(
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]:   File "/home/pi/RASCSI/src/web/venv/lib/python3.9/site-packages/flask/templating.py", line 147, in render_template
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]:     return _render(
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]:   File "/home/pi/RASCSI/src/web/venv/lib/python3.9/site-packages/flask/templating.py", line 128, in _render
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]:     rv = template.render(context)
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]:   File "/home/pi/RASCSI/src/web/venv/lib/python3.9/site-packages/jinja2/environment.py", line 1304, in render
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]:     self.environment.handle_exception()
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]:   File "/home/pi/RASCSI/src/web/venv/lib/python3.9/site-packages/jinja2/environment.py", line 925, in handle_exception
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]:     raise rewrite_traceback_stack(source=source)
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]:   File "/home/pi/RASCSI/src/web/templates/index.html", line 1, in top-level template code
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]:     {% extends "base.html" %}
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]:   File "/home/pi/RASCSI/src/web/templates/base.html", line 60, in top-level template code
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]:     <span style="display: inline-block; width: 100%; color: white; background-color: green; text-align: center; vertical-align: center; font-family: Arial, Helvetica, sans-serif;">{{ _("Web Interface Authentication Disabled") }} &#8211; {{ _("See <a href=\"%(url)\" target=\"_blank\">Wiki</a> for more information", url="https://github.com/akuker/RASCSI/wiki/Web-Interface#enable-authentication") }}</span>
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]:   File "/home/pi/RASCSI/src/web/venv/lib/python3.9/site-packages/jinja2/ext.py", line 168, in _gettext_alias
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]:     return __context.call(__context.resolve("gettext"), *args, **kwargs)
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]:   File "/home/pi/RASCSI/src/web/venv/lib/python3.9/site-packages/jinja2/ext.py", line 180, in gettext
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]:     return rv % variables  # type: ignore
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]:   File "/home/pi/RASCSI/src/web/venv/lib/python3.9/site-packages/markupsafe/__init__.py", line 109, in __mod__
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]:     return self.__class__(super().__mod__(arg))
Dec 26 11:33:20 raspberrypi RASCSIWEB[402089]: ValueError: unsupported format character '"' (0x22) at index 19
```

It seems there is something off in line 60 of base.html.

**Comment by bzeiss** [comment]
I'm not an expert with that jinja templating stuff, but it looks to me as if the placeholder might be missing an s like %()s at the end.
```
diff --git a/src/web/templates/base.html b/src/web/templates/base.html
index 1bf7bf0..5465227 100644
--- a/src/web/templates/base.html
+++ b/src/web/templates/base.html
@@ -57,7 +57,7 @@
         </span>
         {% endif %}
         {% else %}
-        <span style="display: inline-block; width: 100%; color: white; background-color: green; text-align: center; vertical-align: center; font-family: Arial, Helvetica, sans-serif;">{{ _("Web Interface Authentication Disabled") }} &#8211; {{ _("See <a href=\"%(url)\" target=\"_blank\">Wiki</a> for more information", url="https://github.com/akuker/RASCSI/wiki/Web-Interface#enable-authentication") }}</span>
+        <span style="display: inline-block; width: 100%; color: white; background-color: green; text-align: center; vertical-align: center; font-family: Arial, Helvetica, sans-serif;">{{ _("Web Interface Authentication Disabled") }} &#8211; {{ _("See <a href=\"%(url)s\" target=\"_blank\">Wiki</a> for more information", url="https://github.com/akuker/RASCSI/wiki/Web-Interface#enable-authentication") }}</span>
         {% endif %}
     <table width="100%">
         <tbody>
```

**Comment by bzeiss** [comment]
After that, I haven't managed to cause another internal server error. I have also switched an tested with Swedish a bit.

**Comment by rdmark** [comment]
@bzeiss Thanks for testing! And yes, a formatting operator is indeed mandatory, with "s" meaning to interpret it as a string. See https://docs.python.org/2/library/stdtypes.html#string-formatting-operations

File: src/web/translations/sv/LC_MESSAGES/messages.po [hunk 1]
```diff
@@ -0,0 +1,1127 @@
...
+"Last-Translator: Daniel Markstedt <markstedt@gmail.com>\n"
+"Language: sv\n"
+"Language-Team: N/A\n"
+"Plural-Forms: nplurals=2; plural=(n != 1)\n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=utf-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+"Generated-By: Babel 2.9.1\n"
+
+#: file_cmds.py:186
```
Comment (akuker) [inline src/web/translations/sv/LC_MESSAGES/messages.po:21]: @rdmark Are these source files and line number necessary? Do they need to be maintained?

Comment (rdmark) [reply src/web/translations/sv/LC_MESSAGES/messages.po:21]: Good question! The source file and line number stuff are generated by gettext msgfmt (through the pybabel wrapper). So there's no need to maintain that data by hand. As a translator, the only part that you should touch are the ```msgstr``` lines. And, sometimes removing 'fuzzy' flags on translations after the mother string has changed.

**Review by akuker — APPROVED** [review]

**Pull request merged**